test: update tests to align with Jest version 30 (#563)
all-contributors/all-contributors-bot: DEVELOPMENT.md

# Development Guide

This guide explains how the All Contributors app is built, how to run and test it locally, and how to keep your local environment aligned with CI.

## Quick Start

This guide assumes that you have `nvm` installed to manage Node versions. It also assumes that you are working in a fork that has been cloned locally of the 

> all-contributors/app repository.

Author: ```suggestion
This guide assumes that you have `nvm` installed to manage Node versions. It also assumes that you are working in a cloned fork of the `all-contributors/app` GitHub repository.
```

### Environment setup

To begin, be sure that you are using node version 22 (what we are running in CI), install all dependencies using the package-lock.json lockfile.

```console

# Use Node 22 (same as CI)
nvm install 22
nvm use 22

# Install dependencies specified in the package-lock.json lockfile
npm ci
```

### Test suite

Next, you can run the test suite using npm.

```sh
# Run the tests
npm test
```

We use Jest as our test runner.

> If Jest, what we use to run tests has been updated, you may have to update the Jest snapshots from time to time. To do that run:

Reviewer: [Docs] Snapshots can update from more than just Jest updating. Often when I'm working in a codebase that uses snapshot testing I find myself using `-u` a lot. How about generalizing it?

```suggestion
To update snapshots, which can need to happen when tests change and/or Jest is updated, run:
```

Author: Thank you! @JoshuaKGoldberg I'm new to JS and JEST, so I'm learning here. But I understand the concept of snapshots. Please keep these types of suggestions coming!
